### PR TITLE
Implement actions and reducers for movies

### PR DESCRIPTION
**Pull request by KozlovaAngelika**

### CODE REVIEW

File: src/components/Loader/styles.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const styles = StyleSheet.create({
+  container: {
+    position: 'absolute',
+    top: 0,
+    bottom: 0,
+    left: 0,
+    right: 0,
+    justifyContent: 'center',
+    alignItems: 'center',
+    flexGrow: 1,
```
Comment (brusnev-egor) [inline src/components/Loader/styles.ts:13]: flex with position absolute?

File: src/screens/Home/index.tsx [hunk 2]
```diff
@@ -1,37 +1,73 @@
...
+      return <Notice isError={false} message={emptyRequest} />;
+    }
+    if (isLoading) {
+      return <Loader />;
+    }
+    if (error) {
+      return <Notice isError message={errorMessage} />;
+    }
+    if (data?.length === 0) {
+      return <Notice isError={false} message={noResultsMessage} />;
```
Comment (brusnev-egor) [inline src/screens/Home/index.tsx:57]: you can remove isError prop from there

File: src/store/index.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+});
+
+const api = axios.create({
+  headers: {
+    contentType: 'application/json',
+  },
+  baseURL: API_URL,
+});
+
+export type State = ReturnType<typeof rootState.getState>;
```
Comment (brusnev-egor) [inline src/store/index.ts:18]: better to rename it to `RootState`

File: src/store/movies/actions.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  type: types.CLEAR_SEARCH_RESULTS,
+});
+export const searchMovies = (
+  value: string,
+): RootThunkAction<SearchMoviesActions> => (dispatch, getState, api) => {
+  dispatch(searchMoviesStarted());
+  api
+    .get<SearchMovieResponse>(`SearchMovie/${API_KEY}/${value}`)
+    .then((res) => {
+      const { errorMessage, results } = res.data;
```
Comment (brusnev-egor) [inline src/store/movies/actions.ts:33]: .

File: src/store/movies/reducers.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
+/* eslint-disable @typescript-eslint/default-param-last */
+/* eslint-disable import/prefer-default-export */
```
Comment (brusnev-egor) [inline src/store/movies/reducers.ts:2]: remove it and use default export for reducer

File: src/store/movies/selectors.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import { State } from '..';
+
+export const selectMovies = (state: State): Movie[] | null => state.movies.data;
```
Comment (brusnev-egor) [inline src/store/movies/selectors.ts:3]: return type here should be without null. you can workaround it with `... || []`

File: src/store/movies/reducers.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        loading: false,
+        error: action.payload,
+        data: null,
+      };
+    case types.CLEAR_SEARCH_RESULTS:
+      return {
+        ...state,
+        loading: false,
+        error: null,
+        data: null,
```
Comment (brusnev-egor) [inline src/store/movies/reducers.ts:42]: [] instead of null

File: src/store/movies/reducers.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        loading: false,
+        error: null,
+        data: action.payload,
+      };
+    case types.SEARCH_MOVIES_FAIL:
+      return {
+        ...state,
+        loading: false,
+        error: action.payload,
+        data: null,
```
Comment (brusnev-egor) [inline src/store/movies/reducers.ts:35]: []

File: src/store/movies/reducers.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  state: MoviesState = initialState,
+  action: SearchMoviesActions,
+): MoviesState => {
+  switch (action.type) {
+    case types.SEARCH_MOVIES_STARTED:
+      return {
+        ...state,
+        loading: true,
+        error: null,
+        data: null,
```
Comment (brusnev-egor) [inline src/store/movies/reducers.ts:21]: []

File: src/components/SearchBar/index.tsx [hunk 10]
```diff
@@ -2,9 +2,9 @@ import React from 'react';
 import { TextInput, View } from 'react-native';
 import { Icon } from 'react-native-elements';
 import { useTranslation } from 'react-i18next';
-import { COLORS } from '../../utils/constants';
+import { COLORS } from 'utils/constants';
 import styles from './styles';
-import '../../utils/i18n';
+import 'utils/i18n';
```
Comment (brusnev-egor) [inline src/components/SearchBar/index.tsx:7]: remove this import

File: src/types/helper.d.ts [hunk 11]
```diff
@@ -0,0 +1,7 @@
+declare module '*.png';
+
+interface AsyncState {
+  loading: boolean;
+  error: Error | null;
+  data: Movie[];
```
Comment (brusnev-egor) [inline src/types/helper.d.ts:6]: please, use generic here

File: src/screens/Home/index.tsx [hunk 12]
```diff
@@ -1,37 +1,73 @@
...
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const data: Movie[] | null = useSelector(selectMovies);
+  const isLoading: boolean = useSelector(selectLoadingStatus);
+  const error: Error | null = useSelector(selectError);
   const [searchValue, setSearchValue] = useState('');
-  const [dataForDisplay, setDataForDisplay] = useState(data);
+  const errorMessage = t('error');
+  const emptyRequest = t('emptyRequestNotice');
+  const noResultsMessage = t('noResults');
```
Comment (brusnev-egor) [inline src/screens/Home/index.tsx:27]: it is not necessary to create consts for translations, but let's leave it as is

File: src/screens/Home/index.tsx [hunk 13]
```diff
@@ -1,37 +1,69 @@
...
+import Loader from 'components/Loader';
+import { clearSearchResults, searchMovies } from 'store/movies/actions';
+import Notice from 'components/Notice';
 import styles from './styles';
 
 const Home: React.FunctionComponent = () => {
-  const data: Movie[] = useSelector((state: MovieState) => state.movies);
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const data: Movie[] | null = useSelector(selectMovies);
```
Comment (brusnev-egor) [inline src/screens/Home/index.tsx:21]: type without null

**Review by brusnev-egor — APPROVED** [review]

File: src/store/movies/actions.ts [hunk 14]
```diff
@@ -0,0 +1,48 @@
+import { API_KEY } from 'react-native-dotenv';
+import { RootThunkAction } from '..';
```
Comment (i-nemets) [inline src/store/movies/actions.ts:2]: ?

**Review by i-nemets — APPROVED** [review]

**Pull request merged**